src/einsteinpy/metric/kerr.py

```diff
@@ -1,72 +1,402 @@
+    def _g_cov_ks(self, x_vec):
         """
-        Constructor
+        Returns Covariant Kerr Metric Tensor \
+        in Kerr-Schild coordinates
+
+        Parameters
+        ----------
+        x_vec : numpy.array
+            Position 4-Vector
+
+        Returns
+        -------
+        NotImplementedError
+            To be implemented after KS Coordinates
+
+        """
+        # To be implemented after KS Coordinates
+        raise NotImplementedError
+
+    def _dg_dx_bl(self, x_vec):
+        """
+        Returns derivative of each Kerr Metric component \
+        w.r.t. coordinates in Boyer-Lindquist Coordinate System
+
+        Parameters
+        ----------
+        x_vec : numpy.array
+                Position 4-Vector
+
+        Returns
+        -------
+        dgdx : ~numpy.array
+            Array, containing derivative of each Kerr Metric \
+            component w.r.t. coordinates \
+            in Boyer-Lindquist Coordinate System
+            Numpy array of shape (4,4,4)
+            dgdx[0], dgdx[1], dgdx[2] & dgdx[3] contain \
+            derivatives of metric w.r.t. t, r, theta & phi respectively
+
+        """
+        r, th = x_vec[1], x_vec[2]
+        r_s, M, a, c2 = self.sch_rad, self.M, self.a, _c ** 2
+        alpha = BaseMetric.alpha(M, a)
+        sg, dl = BaseMetric.sigma(r, th, M, a), BaseMetric.delta(r, M, a)
+
+        dgdx = np.zeros(shape=(4, 4, 4), dtype=float)
+
+        # Metric is invariant on t & phi
+        # Differentiation of metric wrt r
+        def due_to_r():
+            nonlocal dgdx
+            dsdr = 2 * r
+            dddr = 2 * r - r_s
+            tmp = (r_s * (sg - r * dsdr) / sg) * (1 / sg)
+            dgdx[1, 0, 0] = -1 * tmp
+            dgdx[1, 1, 1] = (-1 / c2) * (dsdr - (sg * (dddr / dl))) / dl
+            dgdx[1, 2, 2] = (-1 / c2) * dsdr
+            dgdx[1, 3, 3] = (
+                (-1 / c2)
+                * (2 * r + (alpha ** 2) * (np.sin(th) ** 2) * tmp)
+                * (np.sin(th) ** 2)
+            )
+            dgdx[1, 0, 3] = dgdx[1, 3, 0] = (1 / _c) * (alpha * (np.sin(th) ** 2) * tmp)
+
+        # Differentiation of metric wrt theta
+        def due_to_theta():
+            nonlocal dgdx
+            dsdth = -2 * (alpha ** 2) * np.cos(th) * np.sin(th)
+            tmp = (-1 / sg) * r_s * r * dsdth / sg
+            dgdx[2, 0, 0] = -1 * tmp
+            dgdx[2, 1, 1] = (-1 / c2) * (dsdth / dl)
+            dgdx[2, 2, 2] = (-1 / c2) * dsdth
+            dgdx[2, 3, 3] = (-1 / c2) * (
+                2 * np.sin(th) * np.cos(th) * ((r ** 2) + (alpha ** 2))
+                + tmp * (alpha ** 2) * (np.sin(th) ** 4)
+                + (4 * (np.sin(th) ** 3) * np.cos(th) * (alpha ** 2) * r * r_s / sg)
+            )
+            dgdx[2, 0, 3] = dgdx[2, 3, 0] = (alpha / _c) * (
+                (np.sin(th) ** 2) * tmp + (2 * np.sin(th) * np.cos(th) * r_s * r / sg)
+            )
+
+        due_to_r()
+        due_to_theta()
```

Comment:
I have simply copied this method from the old API and changed some labels here and there. This function was defined this way in the old API, which is fine, I think. Because these methods have to be hardcoded. My guess is, calculations for the derivatives were done by hand or perhaps taken from some book and then transferred to code. There is no other way to define or use these functions anywhere else, as say, utility functions. So, defining these methods as inner functions and making them private (`_`) should be perfectly fine. Is there a usecase, that I seem to be missing? @rishi-s8